templates/InstallChocolatey.ps1.erb

```diff
@@ -38,9 +38,19 @@ param (
   [string]$file
  )
   Write-Output "Downloading $url to $file"
-  $downloader = new-object System.Net.WebClient
+  $downloader = New-Object System.Net.WebCLient
+<% if @proxy_server -%>
+  $proxy = New-Object System.Net.WebProxy("<% @proxy_server -%>:<% @proxy_port -%>", $true)
+  <%- if @proxy_username -%>
+  $passwd = ConvertTo-SecureString "<% @proxy_password -%>" -AsPlainText -Force; ## Website credentials
+  $proxy.Credentials = New-Object System.Management.Automation.PSCredential ("<% @proxy_username -%>", $passwd);
+  $downloader.UseDefaultCredentials = $true
+  <%- end -%>
+  $downloader.Proxy = $proxy
```

Comment:
This looks very similar to what I just released at https://github.com/chocolatey/choco/blob/2fbecf671dd6a2bfc54ee2c6a97bb152649d340c/src/chocolatey.resources/helpers/functions/Get-WebFile.ps1#L40-L52